src/otx/algorithms/common/adapters/mmcv/hooks/recording_forward_hook.py

```diff
@@ -238,3 +239,122 @@ def _get_mosaic_feature_map(self, feature_map: torch.Tensor, c: int, h: int, w:
                     mosaic_feature_map_mask[k, :, i, j] = torch.ones(c).to(feature_map.device)
             mosaic_feature_map = feature_map_repeated * mosaic_feature_map_mask
         return mosaic_feature_map
+
+
+class ViTReciproCAMHook(BaseRecordingForwardHook):
```

Comment:
Could you add unit test for this class?

Reply:
It is there, check out https://github.com/openvinotoolkit/training_extensions/blob/ceb0ffa8e7c436406578ae2d5238ec67a72e6626/tests/unit/algorithms/classification/test_xai_classification_validity.py#L51

Does it look good?

Reply:
Yes, it checks the function in some degree, but this function has a quite complex code. so, how about to add a little more unit test?

Reply:
I added some unit tests, please take a look